clients/src/main/java/org/apache/kafka/clients/ConnectionState.java

```diff
@@ -16,14 +16,21 @@
  */
 package org.apache.kafka.clients;
 
+import java.util.EnumSet;
+
 /**
  * The states of a node connection
  *
  * DISCONNECTED: connection has not been successfully established yet
  * CONNECTING: connection is under progress
  * CHECKING_API_VERSIONS: connection has been established and api versions check is in progress. Failure of this check will cause connection to close
  * READY: connection is ready to send requests
+ * AUTHENTICATION_FAILED: connection failed due to an authentication error
  */
 public enum ConnectionState {
-    DISCONNECTED, CONNECTING, CHECKING_API_VERSIONS, READY
+    DISCONNECTED, CONNECTING, CHECKING_API_VERSIONS, READY, AUTHENTICATION_FAILED;
+
+    public static EnumSet<ConnectionState> disconnectedStates() {
```

Comment:
Another option is to add an `isDisconnected` field. Then the check is just `state.isDisconnected` instead of `disconnectedStates().contains(state)`.